### PR TITLE
add explicit RegistryClient.Query type

### PR DESCRIPTION
**Pull request by pdowler**
remove local and short hostname mangling
remove domainMatch logic

### CODE REVIEW

File: cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java [hunk 1]
```diff
@@ -342,54 +311,38 @@ public void testGetServiceURLMatchDomain() {
...
-            Assert.assertEquals("Wrong URL.",
+            Assert.assertEquals("default lookup",
                                 "https://mysite.com/services/mygreatservice",
                                 capabilitiesURL.toExternalForm());
+            
+            URL explicitCapURL = registryClient.getAccessURL(RegistryClient.Query.CAPABILITIES, resourceID);
+            Assert.assertEquals("explicit lookup", capabilitiesURL, explicitCapURL);
+            
+            try {
+                registryClient.getAccessURL(RegistryClient.Query.APPLICATIONS, resourceID);
```
Comment (hjeeves) [inline cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java:342]: I see.. it looks like this test is designed to fail. (no applications cache file written above, only capabilities.) Why? Is it making sure the behaviour of getAccessURL has changed?  

There's no need to have an explicit positive test for RegistryClient.Query.APPLICATIONS?

**Review by hjeeves — CHANGES_REQUESTED** [review]

File: cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java [hunk 2]
```diff
@@ -342,54 +311,38 @@ public void testGetServiceURLMatchDomain() {
...
-                    new RegistryClient(new URL("https://www.mysite.com/registry/registry-entries"));
-            final File cacheFile =
-                    new File(new File(System.getProperty("java.io.tmpdir") + fileSeparator
-                                              + System.getProperty("user.name") + fileSeparator + "cadc-registry-1.4"),
-                             "registry-entries");
-            final FileWriter fileWriter = new FileWriter(cacheFile);
+            RegistryClient registryClient = new RegistryClient();
+            String cache = RegistryClient.Query.CAPABILITIES.getValue();
+            File cacheDir = new File(System.getProperty("java.io.tmpdir") + fileSeparator
+                                              + System.getProperty("user.name") + fileSeparator + "cadc-registry-1.4");
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java:323]: It is reflecting the last time the cache structure changed and this is till compat.

[the remark this answers is not in the document]
Comment (pdowler) [reply cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java:323]: oh, that should use the constant... cut&paste from another test.

File: cadc-registry/src/main/java/ca/nrc/cadc/reg/client/RegistryClient.java [hunk 3]
```diff
@@ -133,65 +114,46 @@ public class RegistryClient {
...
+            this.value = s;
+        }
+        
+        public String getValue() {
+            return value;
+        }
+    }
+    
     // version the cache dir so we can increment when we have incompatible cache structure
     static final String CONFIG_CACHE_DIR = "cadc-registry-1.4";
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cadc-registry/src/main/java/ca/nrc/cadc/reg/client/RegistryClient.java:135]: that's the last version where the cache structure changed so this marks compat vs incompat cache structure... so no change in this version

File: cadc-registry/src/main/java/ca/nrc/cadc/reg/client/RegistryClient.java [hunk 4]
```diff
@@ -201,95 +163,78 @@ private String extractFileName(final URL resourceCapsURL) {
...
+        URL queryURL = new URL(regBaseURL + "/" + queryName.getValue());
+        CachingFile cachedCapSource = new CachingFile(queryCacheFile, queryURL);
         String map = cachedCapSource.getContent();
         InputStream mapStream = new ByteArrayInputStream(map.getBytes(StandardCharsets.UTF_8));
         MultiValuedProperties mvp = new MultiValuedProperties();
         try {
             mvp.load(mapStream);
         } catch (Exception e) {
-            throw new RuntimeException("failed to load capabilities source map from " + resourceCapsURL, e);
+            throw new RuntimeException("failed to load capabilities source map from " + regBaseURL, e);
```
[the remark this answers is not in the document]
Comment (pdowler) [reply cadc-registry/src/main/java/ca/nrc/cadc/reg/client/RegistryClient.java:224]: fixing up error messaging in getAccessURL()

File: cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java (hunk 1 above)
Comment (pdowler) [reply cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java:342]: yes, supposed to fail becauyse wrong query; adding an Assert.fail to make test correct.

also adding a positive applications test

**Review by hjeeves — APPROVED** [review]

**Pull request merged**